CMakeLists.txt

```diff
@@ -0,0 +1,225 @@
+    multi-glob.cc
+    perf.cc
+    strerror.cc
+    tar.cc
+    uuid.cc
+    elf/arch-arm32.cc
+    elf/arch-arm64.cc
+    elf/arch-i386.cc
+    elf/arch-riscv64.cc
+    elf/arch-x86-64.cc
+    elf/cmdline.cc
+    elf/dwarf.cc
+    elf/gc-sections.cc
+    elf/icf.cc
+    elf/input-files.cc
+    elf/input-sections.cc
+    elf/linker-script.cc
+    elf/lto.cc
+    elf/main.cc
+    elf/mapfile.cc
+    elf/output-chunks.cc
+    elf/passes.cc
+    elf/relocatable.cc
+    elf/subprocess.cc
+    macho/arch-arm64.cc
+    macho/arch-x86-64.cc
+    macho/cmdline.cc
+    macho/dead-strip.cc
+    macho/input-files.cc
+    macho/input-sections.cc
+    macho/lto.cc
+    macho/main.cc
+    macho/mapfile.cc
+    macho/output-chunks.cc
+    macho/tapi.cc
+    macho/yaml.cc
+    third-party/rust-demangle/rust-demangle.c)
+
+# Create the ld and ld64 symlinks required for testing
+add_custom_command(
+    TARGET mold POST_BUILD
+    COMMAND ${CMAKE_COMMAND} -E create_symlink mold ld
+    WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}
+    COMMENT "Creating ld symlink"
+    VERBATIM)
+add_custom_command(
+    TARGET mold POST_BUILD
+    COMMAND ${CMAKE_COMMAND} -E create_symlink mold ld64
+    WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}
+    COMMENT "Creating ld64 symlink"
+    VERBATIM)
+
+include(CTest)
+
+set(MOLD_TESTS_PATH ${CMAKE_CURRENT_LIST_DIR}/test/)
+
+if(CMAKE_SYSTEM_NAME MATCHES Darwin)
+    string(APPEND MOLD_TESTS_PATH "macho")
+else()
+    string(APPEND MOLD_TESTS_PATH "elf")
+endif()
+
+string(APPEND MOLD_TESTS_PATH "/*")
+FILE(GLOB MOLD_TESTS ${MOLD_TESTS_PATH})
+
+foreach(MOLD_TEST IN LISTS MOLD_TESTS)
+    add_test(NAME ${MOLD_TEST} COMMAND ${MOLD_TEST})
+endforeach()
+
+# FIXME(cwasser): This is just a convenience ctest target,
+# which ensures that mold is build. This should ideally be fixed in CMake.
+include(ProcessorCount)
+ProcessorCount(MOLD_NPROC)
+add_custom_target(check COMMAND ${CMAKE_CTEST_COMMAND} -j${MOLD_NPROC}
+    DEPENDS mold USES_TERMINAL)
+
+install(TARGETS mold DESTINATION ${CMAKE_INSTALL_BINDIR})
+install(TARGETS mold-wrapper DESTINATION lib/mold)
+install(CODE "\
+    file(REAL_PATH \$\{CMAKE_INSTALL_PREFIX\} INSTALL_PREFIX)
+    file(RELATIVE_PATH\
+        REL_SYMLINK\
+        \$\{INSTALL_PREFIX\}/${CMAKE_INSTALL_BINDIR}/mold\
+        \$\{INSTALL_PREFIX\}/${CMAKE_INSTALL_LIBEXECDIR}/mold)\n\
+    execute_process(\
+        COMMAND ${CMAKE_COMMAND} -E make_directory\
+        \$\{INSTALL_PREFIX\}/${CMAKE_INSTALL_LIBEXECDIR}/mold)\n\
+    execute_process(\
+        COMMAND ${CMAKE_COMMAND} -E create_symlink\
+        \$\{REL_SYMLINK\}\
+        \$\{INSTALL_PREFIX\}/${CMAKE_INSTALL_LIBEXECDIR}/mold/ld)\n\
+    execute_process(\
+        COMMAND ${CMAKE_COMMAND} -E create_symlink\
+        mold\
+        \$\{INSTALL_PREFIX\}/${CMAKE_INSTALL_BINDIR}/ld.mold)\n\
+    execute_process(\
+        COMMAND ${CMAKE_COMMAND} -E create_symlink\
+        mold\
+        \$\{INSTALL_PREFIX\}/${CMAKE_INSTALL_BINDIR}/ld64.mold)")
```

Comment:
Only `$` needs escaping, but see above. Symlinks aren't portable. This is also a packaging concern that should be done using CPack.

Reply:
Yeah I'm not happy about this install code either. It's just the most direct port of the existing Makefile installation step I could think of. Is there any better option how I could do this?

Reply:
> This is also a packaging concern that should be done using CPack.

How would I do this using CPack?

Reply:
[CPACK_PRE_BUILD_SCRIPTS](https://cmake.org/cmake/help/latest/module/CPack.html#variable:CPACK_PRE_BUILD_SCRIPTS) allows you to run arbitrary code in the staging directory. I might just make an example for this later.

Reply:
Thanks! Will take a look.

Reply:
Actually, this is not necessary on the mold side. Package maintainers can just do this in their install scripts. I tried looking at the [Debian packager script](https://salsa.debian.org/pkg-llvm-team/mold/-/blob/master/debian/rules) and it's just a makefile calling debhelper. I'm actually curious about a package maintainer's opinion on this. Since I was looking at Debian's package, hopefully @sylvestre can say something here.

The question here would be: is documenting the fact that these symlinks might be necessary enough for package maintainers? According to the [README](https://github.com/rui314/mold#how-to-use) these symlinks are only needed for GCC <12.1.0, if I understand things correctly.